mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -580,4 +401,170 @@ impl Nexus {
             .for_each(drop);
         blockcnt
     }
+
+    /// Starts a rebuild task in the background
+    pub async fn start_rebuild_rpc(&mut self, name: &str) -> Result<(), Error> {
+        if let Err(e) = self.start_rebuild(name).await {
+            Err(e)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Starts a rebuild task and returns a receiver channel
+    /// which can be used to await the rebuild completion
+    pub async fn start_rebuild(
+        &mut self,
+        name: &str,
+    ) -> Result<Receiver<RebuildState>, Error> {
+        trace!("{}: start rebuild request for {}", self.name, name);
+
+        let source = match self
+            .children
+            .iter_mut()
+            .find(|c| c.state == ChildState::Open)
+        {
+            Some(child) => child.name.clone(),
+            None => {
+                return Err(Error::OpenChildNotFound {
+                    name: self.name.clone(),
+                })
+            }
+        };
+
+        if let Some(dst_child) =
```

Comment:
Suggestion: We could lay this code out in the same way as the above lines (422-433) for consistency.

Reply:
I actually added a handy fn get_child_by_name but forgot to use it